### DISCUSSION TITLE
Elevante GITHUB_TOKEN permissions in pull_request_review workflow

### DISCUSSION THREAD

**Discussion by carolynvs** [community/community#26651] [Actions]
<p>Is it possible to elevate the permissions for the default GITHUB_TOKEN in a pull_request_review workflow? I have a workflow that assigns write access to the token for the issue_comment event type, but the same workflow only has read access when triggered from a pull_request_review. The pull request originates from a fork.</p>
<pre><code class="lang-auto">name: "Prow github actions"
on:
  issue_comment:
    types: [created,edited]
  pull_request_review:
    types: [submitted,edited]
permissions:
  issues: write
  pull-requests: write

jobs:
  execute:
    runs-on: ubuntu-latest
    steps:
      - uses: carolynvs/prow-github-actions@trigger-from-review
        with:
          prow-commands: |
            /approve
            /lgtm
          github-token: "${{ secrets.GITHUB_TOKEN }}"
</code></pre>
<p>Elevating permissions works for a pull_request_target event and since pull_request_review also runs from the base ref, and doesn’t use the code from the pull request, I hoped that it would also be safe to elevate permissions.</p>
<p>I can’t tell if I’m doing something wrong, there’s a bug or gap, or if I’m misunderstanding the pull_request_target event and it doesn’t actually use the base ref instead of the pr ref?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="carolynvs" data-post="1" data-topic="251756">
<div class="title">
<div class="quote-controls"></div>
 carolynvs:</div>
<blockquote>
<p>since pull_request_review also runs from the base ref, and doesn’t use the code from the pull request</p>
</blockquote>
</aside>
<p>According to the <a href="https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_review"><code>pull_request_review</code> event documentation</a> the <code>pull_request_review</code> event runs on the modified code (note the <code>GITHUB_REF</code>). I assume that’s why the permissions are just as restricted as for <code>pull_request</code>.</p>

**Comment by carolynvs** [comment]
<p>Shoot, yeah I see that I was reading the doc incorrectly now.</p>
<p>Too bad there isn’t a pull_request_review_target event. :smiley:  Maybe one day more *_target events will be available so that people can safely kick off workflows based on pull request activity.</p>

**Comment by slawekjaranowski** [comment]
I have similar issue - I would like to add missing labels, milestones after PR is approved.

**Comment by slawekjaranowski** [comment]
and conncted discussion: #55940
